### ISSUE TITLE
sdk-glacier: ArchiveTransferManager requires AmazonGlacierClient instead of AmazonGlacier

### ISSUE THREAD

**Issue by kac-** [aws/aws-sdk-java#1004]
*ArchiveTransferManager* in **all** its constructors requires *AmazonGlacier***Client**, [i.e. here](aws-java-sdk-glacier/src/main/java/com/amazonaws/services/glacier/transfer/ArchiveTransferManager.java#L165), instead of *AmazonGlacier* - an interface implemented by *AmazonGlacier***Client**.
```java
private final AmazonGlacier glacier;

...

public ArchiveTransferManager(AmazonGlacierClient glacier, AWSCredentialsProvider credentialsProvider, ClientConfiguration clientConfiguration) {
        this.credentialsProvider = credentialsProvider;
        this.clientConfiguration = clientConfiguration;
        this.glacier = glacier;
        this.sns = null;
        this.sqs = null;
}
```

**Comment by dagnir** [comment]
Hi @kac-, thanks for pointing this out. I'll work on getting a fix out.

**Comment by dagnir** [comment]
Just an update, I have a fix for this upstream and it should be available in the next release. Thanks!

**Comment by dagnir** [comment]
This is fixed in [1.11.91](https://github.com/aws/aws-sdk-java/blob/a3168a27c7cda03aaf22cf9eed894faacb98c7e1/CHANGELOG.md). Note that rather than adding a new constructor, we introduced [`ArchiveTransferManagerBuilder`](https://github.com/aws/aws-sdk-java/blob/a3168a27c7cda03aaf22cf9eed894faacb98c7e1/aws-java-sdk-glacier/src/main/java/com/amazonaws/services/glacier/transfer/ArchiveTransferManagerBuilder.java) which allows you to specify any implementation of the clients. Resolving this issue.
